### PR TITLE
feat: 🎸 API managed groups mock mirage

### PR DESCRIPTION
**Pull request by calcaide**
Add API managed groups mock in mirage

### CODE REVIEW

File: addons/api/mirage/factories/managed-groups.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import permissions from '../helpers/permissions';
+
+const types = ['password', 'OIDC'];
+
+export default factory.extend({
+  id: (i) => `managed-groups-id-${i}`,
+  type: (i) => types[i % types.length],
+
+  authorized_actions: () =>
+    permissions.authorizedActionsFor('managed-groups') || [
```
Comment (calcaide) [inline addons/api/mirage/factories/managed-groups.js:11]: @DhariniJeeva I want to make sure: from the permissions and capabilities POV, I am missing something else?

File: addons/api/mirage/scenarios/default.js [hunk 2]
```diff
@@ -44,6 +44,13 @@ export default function (server) {
   server.createList('role', 1, { scope: globalScope }, 'withPrincipals');
   server.createList('role', 5, { scope: orgScope }, 'withPrincipals');
 
+  server.createList(
```
Comment (calcaide) [inline addons/api/mirage/scenarios/default.js:47]: @randallmorey 
Not sure if this is the right location for this piece of code.

File: addons/api/mirage/serializers/managed-groups.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export default ApplicationSerializer.extend({
+  modelName: 'managed-groups',
+
+  _hashForModel() {
+    const json = ApplicationSerializer.prototype._hashForModel.apply(
+      this,
+      arguments
+    );
+    return json;
```
Comment (calcaide) [inline addons/api/mirage/serializers/managed-groups.js:11]: @randallmorey 
Should I add here something like `json.auth_method_id=model.authMethodId`?

File: addons/api/mirage/factories/managed-groups.js (hunk 1 above)
Comment (DhariniJeeva) [reply addons/api/mirage/factories/managed-groups.js:11]: looks good to me. wondering if we need a list/create here?

File: addons/api/mirage/config.js [hunk 4]
```diff
@@ -514,6 +514,14 @@ export default function () {
   this.del('/credential-libraries/:id');
   this.patch('/credential-libraries/:id');
 
+  // managed-groups
+  this.get(
```
Comment (calcaide) [inline addons/api/mirage/config.js:518]: For now just adding the `get` method we will need for the list. Later on I will add more methods.

File: addons/api/mirage/factories/managed-groups.js (hunk 1 above)
Comment (calcaide) [reply addons/api/mirage/factories/managed-groups.js:12]: I just added `authorized_collection_actions` with create and list as I saw in other factory. Is this what you were meaning @DhariniJeeva ?

Comment (DhariniJeeva) [reply addons/api/mirage/factories/managed-groups.js:12]: yes, that's what I meant.

File: addons/api/mirage/factories/managed-group.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  authorized_actions: () =>
+    permissions.authorizedActionsFor('managed-group') || [
+      'no-op',
+      'read',
+      'update',
+      'delete',
+    ],
+
+  authorized_collection_actions: () => {
```
Comment (randallmorey) [inline addons/api/mirage/factories/managed-group.js:15]: Managed groups do not have sub collections.

File: addons/api/mirage/generated/factories/managed-group.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Factory } from 'ember-cli-mirage';
+import { random, date, datatype } from 'faker';
+
+/**
+ * GeneratedManagedGroupFactory
+ * A managed group is a resource that represents a collection of accounts.
```
Comment (randallmorey) [inline addons/api/mirage/generated/factories/managed-group.js:6]: It represents a collection of users I thought?

**Review by randallmorey — APPROVED** [review]

File: addons/api/mirage/factories/managed-group.js (hunk 5 above)
Comment (calcaide) [reply addons/api/mirage/factories/managed-group.js:15]: @randallmorey 
So `authorized_collection_actions` does not apply here?

But I should apply `create` and `list` permissions, right?

File: addons/api/mirage/generated/factories/managed-group.js (hunk 6 above)
Comment (calcaide) [reply addons/api/mirage/generated/factories/managed-group.js:6]: I am happy to change it but I grab it [from here](https://www.boundaryproject.io/docs/concepts/domain-model/managed-groups).

File: addons/api/mirage/factories/managed-group.js (hunk 5 above)
Comment (randallmorey) [reply addons/api/mirage/factories/managed-group.js]: To mock permission to `create` and `list` `managed-groups`, add an entry to the `auth-method` factory.  `authorized_collection_actions` defines what permissions the user has on _sub-collections_.  There are no sub-collections under managed groups itself, so it doesn't not need this field.

File: addons/api/mirage/generated/factories/managed-group.js (hunk 6 above)
Comment (randallmorey) [reply addons/api/mirage/generated/factories/managed-group.js:6]: Ah you are correct.  This indicates that `member_ids` is an array of accounts, not users.

Comment (randallmorey) [reply addons/api/mirage/generated/factories/managed-group.js:6]: But double check with BE.

**Pull request merged**